sdk/src/asset_handlers/gif_io.rs

```diff
@@ -1118,19 +1118,26 @@ impl DataSubBlocks {
     }
 
     fn to_decoded_bytes(&self) -> Vec<u8> {
-        // Amount of bytes - (length markers + terminator).
-        let mut bytes = Vec::with_capacity(self.bytes.len() - (self.bytes.len().div_ceil(255) + 1));
-        for chunk in self.bytes.chunks(256) {
-            bytes.extend_from_slice(&chunk[1..]);
+        let mut bytes = Vec::with_capacity(gif_chunks(&self.bytes).map(|c| c.len()).sum());
```

Comment:
If you call gif_chunks once and save the value you can avoid iterating twice.

Reply:
I'll fix this up later. Thanks for the fix!

Reply:
The iterator is super cheap. It's intentionally run twice, because it avoids reallocation that's more expensive.